cmake/compiler/gcc/target_riscv.cmake

```diff
@@ -15,14 +15,10 @@ endif()
 
 if(CONFIG_FPU)
     if(CONFIG_CPU_HAS_FPU_DOUBLE_PRECISION)
-        if(CONFIG_FLOAT_HARD)
-            string(CONCAT riscv_mabi ${riscv_mabi} "d")
-        endif()
+        string(CONCAT riscv_mabi ${riscv_mabi} "d")
         string(CONCAT riscv_march ${riscv_march} "fd")
     else()
-        if(CONFIG_FLOAT_HARD)
-            string(CONCAT riscv_mabi ${riscv_mabi} "f")
-        endif()
+        string(CONCAT riscv_mabi ${riscv_mabi} "f")
         string(CONCAT riscv_march ${riscv_march} "f")
     endif()
```

Comment:
It cannot be removed, I think.
RISC-V spec allows combination hard-float CPU (ex. -march=rv32imaf) and soft-float ABI (ex. -mabi=ilp32).
In this combination, floating point number are passed on generic registers "aX", move to floating registers "faX" and calculated by HW FPU.

Reply:
OK. The change has now been removed from the PR.
Thank you for bringing this scenario to my attention.